Source/bmalloc/libpas/src/libpas/pas_report_crash.c

```diff
@@ -97,57 +102,103 @@ kern_return_t pas_report_crash_extract_pgm_failure(vm_address_t fault_address, m
 
         addr64_t key = (addr64_t) hash_map_entry->key;
 
-        // Lower PGM Bounds Checking
-        addr64_t bottom = (addr64_t) (key - pgm_metadata->mem_to_waste - pgm_metadata->page_size);
-        addr64_t top = (addr64_t) (key - pgm_metadata->mem_to_waste);
-
-        if ((fault_address >= bottom)
-            && (fault_address < top)) {
-            report->error_type = "out-of-bounds";
-            report->confidence = "high";
-            report->fault_address = fault_address;
-            report->allocation_size = pgm_metadata->allocation_size_requested;
-            return KERN_SUCCESS;
+        if (pgm_metadata->right_align) {
+            report->alignment = "address right-aligned";
+
+            // Right-aligned "Lower PGM OOB" checking
+            addr64_t bottom = (addr64_t) (key - pgm_metadata->mem_to_waste - pgm_metadata->page_size);
+            addr64_t top = (addr64_t) (key - pgm_metadata->mem_to_waste);
+
+            if (pas_fault_address_is_in_bounds(fault_address, bottom, top)) {
```

Comment:
I still feel we have a lot of code duplication here:

Lines 113-117 are duplicated at 140-144, 158-161, and 182-185.
Lines 125-128 are duplicated at 170-173.

We can create functions or macros to help limit the amount of code duplication.